cmake/app/boilerplate.cmake

```diff
@@ -96,8 +96,35 @@ set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${AUTOCONF_H})
 
 include(${ZEPHYR_BASE}/cmake/extensions.cmake)
 
+# Python must be detected before extensions.cmake is used
 find_package(PythonInterp 3.4)
 
+find_system_configuration_directory(dir)
+check_if_directory_is_writeable(${dir} ok)
+if(${ok})
+  set(local_dir ${dir})
+else()
+  # It is not expected that the system configuration directory will
+  # not be writeable, but just in case, we fall back to using the
+  # repo.
+  set(local_dir ${ZEPHYR_BASE})
+endif()
+set(local_cmake ${local_dir}/zephyr/local.cmake)
```

Comment:
Is that file a user or a system/application file? AFAIK configuration files that are maintained by the user are commonly maintained as .<app>rc files and if there are many, they will be in some .<app>rc/ directory. If this is going to .local/<something> they might appear as non-user editable and do not belong to the user.

Reply:
This is a user file.

I was not aware of the .rc convention, I will look into if it makes sense to use it here.

Reply:
Using the term 'rc' would communicate something appropriate about the file, but I don't want to use the suffix .rc as that miscommunicates something about the format of the file. So if we are to use the term 'rc' I would suggest rc.cmake or local.rc.cmake.

```
AFAIK configuration files that are maintained by the user are commonly maintained as .rc files and if there are many, they will be in some .rc/ directory. If this is going to .local/ they might appear as non-user editable and do not belong to the user.
```

AFAIK placing configuration files in ~/.config/$VENDOR is an appropriate location and will have the appropriate permissions. (Initially I supported XDG_CONFIG_HOME in addition to HOME/.config, but found that this complicated things for the user as there were two locations to search for the file. I believe it is safe to not support XDG_CONFIG_HOME as there are so many other applications that also do not support it).